### ISSUE TITLE
Bad `loc` on validation error, if payload represended by one model

### ISSUE THREAD

**Issue by Dock1100** [tiangolo/fastapi#1088]
### Describe the bug

Really like your framework, but there is, indeed, an annoying issue with `loc` on validation error with one object as payload.

### To Reproduce
Code sample
```Python
from typing import List

from fastapi import FastAPI, Body
from pydantic import BaseModel
app = FastAPI()

class NameModel(BaseModel):
    name: str

@app.post("/test", response_model=NameModel)
def test(obj: NameModel, ):  # bad
    return obj

@app.post("/test_embed", response_model=NameModel)
def test(obj: NameModel = Body(..., embed=True)):  # ok
    return obj

@app.post("/test_multiple", response_model=List[NameModel])
def test(obj1: NameModel, obj2: NameModel):  # ok
    return obj1, obj2
```

When you make a request to endpoint (`/test`) with the wrong payload (e.g.: `{}`), it always includes the variable name into error location, despite it has no relation to request.

It makes no sense, moreover, it complicates the logic for error printing on fronted, because they just don't know and not required to know the name of the backend`s internal variable.

```json
{
    "detail": [
        {
            "loc": [
                "body",
                "obj",
                "name"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```
it should be
```json
{
    "detail": [
        {
            "loc": [
                "body",
                "name"
            ],
            "msg": "field required",
            "type": "value_error.missing"
        }
    ]
}
```

With the embedded object (`/test_embed`) or multiple objects (`/test_multiple`), it works as expected, putting the variable name into location, because it should be in the payload.

### Expected behavior

Don't include the variable name into location error, if it is not reflected in schema / not embedded / not expected to be in payload.

### Environment

- OS: macOS
- FastAPI 0.52.0
- Python 3.6.8

**Comment by garry-jeromson** [comment]
We can also observe this behaviour, it caused a bit of head-scratching when trying to diagnose the source of an error between frontend and backend services.

**Comment by m-vellinga** [comment]
Also observed this behaviour when fiddling around with the framework for the first time. The test_embed sample does the job for now so thanks for including that in the issue.

**Comment by tiangolo** [comment]
Thanks for reporting it and for the discussion here everyone! :coffee: 

It was fixed on https://github.com/tiangolo/fastapi/pull/1553 by @amacfie :rocket: 

Released in FastAPI version `0.58.0` :tada:
